toolbox/include/base/ToolboxSingleton.h

```diff
@@ -0,0 +1,142 @@
+     */
+    int numberOfDoFs(const std::string& confKey);
+
+    // GET METHODS
+    // ===========
+
+    /**
+     * Returns the singleton instance to this object.
+     *
+     * @note The singleton stays in memory until the entire matlab is closed. The WBToolbox
+     *       is designed in such a way that assures to reach a clean state when all blocks
+     *       have been terminated.
+     *
+     * @return the singleton instance
+     */
+    static wbt::ToolboxSingleton& sharedInstance();
+
+    // TODO: check if doxygen ref work
+
+    /**
+     * Returns the Configuration object labelled by confKey.
+     * This object is contained into the wbt::RobotInterface object.
+     *
+     * @see ToolboxSingleton::getRobotInterface
+     * @param  confKey The key describing the configuration (name of the Simulink block)
+     * @return         A constant reference to the Configuration object
+     */
+    const Configuration& getConfiguration(const std::string& confKey);
+
+    /**
+    * Returns a \c shared_ptr to the RobotInterface object containing the configuration
+    * labelled by confKey and all the objects used to gather and set robot data
+    *
+    * @param confKey The key describing the configuration (name of the Simulink block)
+    * @return        A \c shared_ptr to the RobotInterface of the requested configuration
+    */
+    const std::shared_ptr<RobotInterface> getRobotInterface(const std::string& confKey);
+
+    /**
+     * Returns a \c shared_ptr to the KinDynComputations object used to perform calculation
+     * on the provided model
+     *
+     * @param confKey The key describing the configuration (name of the Simulink block)
+     * @return        A \c shared_ptr to the iDynTree::KinDynComputations of the requested
+     *                configuration
+     */
+    const std::shared_ptr<iDynTree::KinDynComputations> model(const std::string& confKey);
+
+    // TOOLBOXSINGLETON CONFIGURATION
+    // ==============================
+
+    /*! Saves in the singleton a new configuration \c config.
+     *
+     * If the config is valid and hasn't been already stored, it creates a new entry
+     * in ToolboxSingleton::m_interfaces. If a configuration with matching confKey is found,
+     * if the Configuration object is the same it does nothing, otherwise it overrides it.
+     *
+     * @note Since confKey is the name of the block, the overriding can happen only after
+     *       subsequent compilation in Simulink, when the block's parameters have been
+     *       changed.
+     *
+     * @param  confKey The key describing the configuration (name of the Simulink block)
+     * @param  config  The wbt::Configuration object parsed from Simulink's parameters
+     * @return         Returns \c true if configure is successful, \c false otherwise
+     * @see            ToolboxSingleton::isKeyValid
+     */
+    bool configure(const std::string& confKey, const Configuration& config);
```

Comment:
as before, I do not like the name `configure`

Reply:
Renamed to `storeConfiguration()`